src/graph.rs

```diff
@@ -199,10 +199,11 @@ fn prepare_commodities_graph_for_validation(
     let demand_node_index = filtered_graph.add_node(GraphNode::Demand);
     for (commodity_id, commodity) in commodities {
         if time_slice_selection.level() == commodity.time_slice_level
-            && commodity
-                .demand
-                .get(&(region_id.clone(), year, time_slice_selection.clone()))
-                .is_some_and(|&v| v > Flow(0.0))
+            && commodity.demand.contains_key(&(
```

Comment:
Actually, the first check is redundant here. If the time slice level is different to the one the commodity has, then the key won't exist either.

```suggestion
if commodity.demand.contains_key(&(
```